drivers/adc/adc_mcux_lpadc.c

```diff
@@ -88,37 +91,88 @@ static int mcux_lpadc_channel_setup(const struct device *dev,
 		return -EINVAL;
 	}
 
+	/* Select ADC CMD register to configure based off channel ID */
+	cmd = &data->cmd_config[channel_cfg->channel_id];
+
+	/* If bit 5 of input_positive is set, then channel side B is used */
+	channel_side = 0x20 & channel_cfg->input_positive;
+	/* Channel number is selected by lower 4 bits of input_positive */
+	channel_num = ADC_CMDL_ADCH(channel_cfg->input_positive);
+
+	LOG_DBG("Channel num: %u, channel side: %c", channel_num,
+		channel_side == 0 ? 'A' : 'B');
+
+	LPADC_GetDefaultConvCommandConfig(cmd);
+
 	if (channel_cfg->differential) {
-		LOG_ERR("Differential channels are not supported");
+		/* Channel pairs must match in differential mode */
+		if ((channel_cfg->input_positive | 0x20) !=
+		    channel_cfg->input_negative) {
+			return -ENOTSUP;
+		}
+#if defined(FSL_FEATURE_LPADC_HAS_CMDL_DIFF) && FSL_FEATURE_LPADC_HAS_CMDL_DIFF
+		if (IS_ENABLED(CONFIG_LPADC_DIFFERENTIAL_MODE_AB)) {
+			cmd->sampleChannelMode =
+				kLPADC_SampleChannelDiffBothSideAB;
+		} else {
+			cmd->sampleChannelMode =
+				kLPADC_SampleChannelDiffBothSideBA;
+		}
+#else
+		cmd->sampleChannelMode = kLPADC_SampleChannelDiffBothSide;
+#endif
+	} else if (channel_side != 0) {
+		cmd->sampleChannelMode = kLPADC_SampleChannelSingleEndSideB;
+	} else {
+		/* Default value for sampleChannelMode is SideA */
+	}
+#if defined(FSL_FEATURE_LPADC_HAS_CMDL_CSCALE) && FSL_FEATURE_LPADC_HAS_CMDL_CSCALE
+	/*
+	 * The true scaling factor used by the LPADC is 30/64, instead of
+	 * 1/2. Select 1/2 as this is the closest scaling factor available
+	 * in Zephyr.
+	 */
+	if (channel_cfg->gain == ADC_GAIN_1_2) {
+		LOG_INF("Channel gain of 30/64 selected");
+		cmd->sampleScaleMode = kLPADC_SamplePartScale;
+	} else if (channel_cfg->gain == ADC_GAIN_1) {
+		cmd->sampleScaleMode = kLPADC_SampleFullScale;
+	} else {
+		LOG_ERR("Invalid channel gain");
 		return -EINVAL;
 	}
-
+#else
 	if (channel_cfg->gain != ADC_GAIN_1) {
 		LOG_ERR("Invalid channel gain");
 		return -EINVAL;
 	}
+#endif
 
 	if (channel_cfg->reference != ADC_REF_EXTERNAL0) {
 		LOG_ERR("Invalid channel reference");
 		return -EINVAL;
 	}
 
+	cmd->channelNumber = channel_num;
 	return 0;
 }
 
 static int mcux_lpadc_start_read(const struct device *dev,
 		 const struct adc_sequence *sequence)
 {
+	const struct mcux_lpadc_config *config = dev->config;
 	struct mcux_lpadc_data *data = dev->data;
+	lpadc_conv_command_config_t tmp;
```

Comment:
to reduce stack size and amount of memory accesses can we just make two variables, one for conversion resolution and one for hardware average, instead of the entire conv_command_config_t structure which is much larger and most of the fields not used here